streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -0,0 +1,254 @@
+    private Metrics metrics;
+
+    private KafkaClient kafkaClient;
+    private StreamsConfig streamsConfig;
+
+    private static final int MAX_INFLIGHT_REQUESTS = 100;
+    private static final long MAX_WAIT_TIME_MS = 30000;
+
+    public StreamsKafkaClient(final StreamsConfig streamsConfig) {
+
+        this.streamsConfig = streamsConfig;
+        final Time time = new SystemTime();
+
+        Map<String, String> metricTags = new LinkedHashMap<>();
+        metricTags.put("client-id", StreamsConfig.CLIENT_ID_CONFIG);
+
+        Metadata metadata = new Metadata(streamsConfig.getLong(StreamsConfig.RETRY_BACKOFF_MS_CONFIG), streamsConfig.getLong(StreamsConfig.METADATA_MAX_AGE_CONFIG));
+        List<InetSocketAddress> addresses = ClientUtils.parseAndValidateAddresses(streamsConfig.getList(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG));
+        metadata.update(Cluster.bootstrap(addresses), time.milliseconds());
+
+        MetricConfig metricConfig = new MetricConfig().samples(streamsConfig.getInt(CommonClientConfigs.METRICS_NUM_SAMPLES_CONFIG))
+                .timeWindow(streamsConfig.getLong(CommonClientConfigs.METRICS_SAMPLE_WINDOW_MS_CONFIG), TimeUnit.MILLISECONDS)
+                .tags(metricTags);
+        List<MetricsReporter> reporters = streamsConfig.getConfiguredInstances(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG,
+                MetricsReporter.class);
+        // TODO: This should come from the KafkaStream
+        reporters.add(new JmxReporter("kafka.streams"));
+        this.metrics = new Metrics(metricConfig, reporters, time);
+
+        ChannelBuilder channelBuilder = ClientUtils.createChannelBuilder(streamsConfig.values());
+
+        Selector selector = new Selector(streamsConfig.getLong(StreamsConfig.CONNECTIONS_MAX_IDLE_MS_CONFIG), this.metrics, time, "kafka-client", channelBuilder);
+
+        kafkaClient = new NetworkClient(
+                selector,
+                metadata,
+                streamsConfig.getString(StreamsConfig.CLIENT_ID_CONFIG),
+                MAX_INFLIGHT_REQUESTS, // a fixed large enough value will suffice
+                streamsConfig.getLong(StreamsConfig.RECONNECT_BACKOFF_MS_CONFIG),
+                streamsConfig.getInt(StreamsConfig.SEND_BUFFER_CONFIG),
+                streamsConfig.getInt(StreamsConfig.RECEIVE_BUFFER_CONFIG),
+                streamsConfig.getInt(StreamsConfig.REQUEST_TIMEOUT_MS_CONFIG), time);
+
+    }
+
+
+    /**
+     * Cretes a new topic with the given number of partitions and replication factor.
+     *
+     * @param internalTopicConfig
+     * @param numPartitions
+     * @param replicationFactor
+     */
+    public void createTopic(final InternalTopicConfig internalTopicConfig, final int numPartitions, final int replicationFactor, final long windowChangeLogAdditionalRetention) {
+        Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
```

Comment:
It would be great if you do, but we can probably do it as another PR if you'd prefer. Up to you